### ISSUE TITLE
Feature Request - Training Progress callback

### ISSUE THREAD

**Issue by jrdzha** [awslabs/autogluon#492]
Is it possible to have a progress callback function with ETA or # completed tasks / # total tasks?

**Comment by Innixma** [comment]
Could you clarify which task you are referring to? Do you mean for Tabular on a model-by-model basis or for generic HPO in terms of HPO trials?

**Comment by jrdzha** [comment]
Hi, I mean on a model by model basis. Just something similar to H2O-3's AutoML, how the user can query the leaderboard, get the current leader and its stats, etc. This is just useful to help track long running sessions and potentially build live graphs or dashboards.

**Comment by Innixma** [comment]
I'll need to think about the best way to incorporate this functionality, but it is definitely do-able and useful. I will update this issue when I make progress in my implementation.

**Comment by arunmk** [comment]
@Innixma thanks for taking up this bug. We are also interested in this and want details from all models including non-leaders while the training is going on. Would that be supported by the fix for this issue?

**Comment by Innixma** [comment]
Yes, this is something that I plan to add as it is a natural aspect of having the callback function.

**Comment by jrdzha** [comment]
Hi @Innixma, just wondering if this is still in the plans? Thanks!

**Comment by Innixma** [comment]
@jrdzha This is planned as a stretch goal for 0.1 release in January, but more likely as a follow-up feature likely to release sometime in late Q1 2020.
